### PR TITLE
feat: add support for custom objectId

### PR DESCRIPTION
**Pull request by martinpfannemueller**
I implemented the custom objectId feature mentioned in #1087 including unit tests. Is there something missing to get this merged?

Thanks!

### CODE REVIEW

**Comment by martinpfannemueller** [comment]
I just merged the changes from the master. Again, locally everything runs just fine.

**Comment by martinpfannemueller** [comment]
Still the same problem. This maybe has to do something with the OS or the development environment then. I am running the tests inside Android Studio on macOS. In general, do you have an idea how we can debug the two failing tests in the CI pipeline?

**Comment by martinpfannemueller** [comment]
I just tried to update the unit tests as it was partly possible to reproduce the failing tests. It definitely has something to do with the Parse instance. Therefore, I tried to clean them up properly now in the new tests at the end.

**Comment by mtrezza** [comment]
> It would definitely help if you could provide some guidelines how the mocking and the different plugins as well as the different states of the mocked objects work together in the tests, e.g., by providing a sample when using the local datastore or when a specific response is expected.

@L3K0V @Jawnnypoo could you do that or should I look into it?

**Comment by Jawnnypoo** [comment]
I don't have context into that, sorry.

**Review by L3K0V — APPROVED** [review]
@mtrezza @martinpfannemueller LGTM

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you please also add doc headers to the new public properties and methods?

**Comment by L3K0V** [comment]
@martinpfannemueller gentle ping when you have some free time if is possible to add some JavaDoc as @mtrezza pleases, so this can be merged. Looking forward to see this merged 🤞

**Comment by martinpfannemueller** [comment]
I just added some JavaDoc to allowCustomObjectId() in the configuration builder and the isAllowCustomObjectId() getter.

File: parse/src/main/java/com/parse/Parse.java [hunk 1]
```diff
@@ -657,6 +669,20 @@ private Builder setLocalDatastoreEnabled(boolean enabled) {
                 return this;
             }
 
+            /**
+             * Enable the possibility to set a custom objectId for your ParseObjects.
```
Comment (mtrezza) [inline parse/src/main/java/com/parse/Parse.java:673]: Could you shorten this to:

> Allow to set a custom objectId for ParseObjects.

**Comment by mtrezza** [comment]
There also seems to be a lint fix needed.

**Comment by martinpfannemueller** [comment]
I updated the JavaDoc and ran spotlessApply

**Comment by mtrezza** [comment]
Thanks, let just see whether the CI passes.

**Review by mtrezza — APPROVED** [review]
Looks good, thanks for this PR!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [2.1.0](https://github.com/parse-community/Parse-SDK-Android/releases/tag/2.1.0)
